### ISSUE TITLE
upsample top index xxx out of range 

### ISSUE THREAD

**Issue by SpecialM2580** [alexgkendall/caffe-segnet#5]
Hi,@kezbreen
I try your code, but got the following error. 
I followed the steps in README.md,step by step,here is the list:

download camvid dataset(701 images) and transform the label by using the script from issue #3 
resized the original pics into [360 480].
generate data.txt
change the path in net.prototxt(first 2 layer)
change the 'dense_softmax_inner_prod' layer in "num_output: 32" (camvid 32 kinds of label)
change the solver.prototxt "solver_mode" to cpu(i used the Geforce 970,it showed error==cudaSuccess(2,0), i thought it was not enough memory ? So I change to cpu)
then I start training...

...
I1002 10:58:40.084039  3306 solver.cpp:42] Solver scaffolding done.
I1002 10:58:40.084084  3306 solver.cpp:250] Solving segnet
I1002 10:58:40.084090  3306 solver.cpp:251] Learning Rate Policy: fixed
I1002 10:58:40.085350  3306 solver.cpp:294] Iteration 0, Testing net (#0)
I1002 10:58:46.428558  3306 solver.cpp:343]     Test net output #0: accuracy = 0.0286227
I1002 10:58:46.428601  3306 solver.cpp:343]     Test net output #1: loss = 3.49524 (\* 1 = 3.49524 loss)
F1002 10:58:58.492833  3306 upsample_layer.cpp:127] upsample top index 481 out of range - check scale settings match input pooling layer's downsample setup
**\* Check failure stack trace: ***
    @     0x7fc354130ea4  (unknown)
    @     0x7fc354130deb  (unknown)
    @     0x7fc3541307bf  (unknown)
    @     0x7fc354133a35  (unknown)
    @     0x7fc359fa814d  caffe::UpsampleLayer<>::Backward_cpu()
    @     0x7fc359f551e9  caffe::Net<>::BackwardFromTo()
    @     0x7fc359f55551  caffe::Net<>::Backward()
    @     0x7fc359f61135  caffe::Solver<>::Step()
    @     0x7fc359f61da5  caffe::Solver<>::Solve()
    @           0x407c7b  train()
    @           0x404ed8  main
    @     0x7fc3509a9a40  (unknown)
    @           0x404a79  _start
    @              (nil)  (unknown)
Aborted (core dumped)
...

I change the batch size to 5 in net.prototxt, the result is 
F1002 10:58:58.492833  3306 upsample_layer.cpp:127] upsample top index 0 out of range - check scale settings match input pooling layer's downsample setup

Please help, i'm new in caffe .Thanks

**Comment by jorgyz** [comment]
@SpecialM2580  Having the same problem here...were you able to figure it out? Thanks

**Comment by jaykay2000** [comment]
@SpecialM2580 
Hi, I ran into the same problem, have you guys figured out how to fix this problem? The funny thing is that by changing the net.prototxt the index "number" changes...

**Comment by jorgyz** [comment]
Interestingly, if you comment out that check in the code and recompile, it runs just fine and the results seem to look as expected.

**Comment by SpecialM2580** [comment]
I'm sorry，I almost forget this issue…It seems that this problem only occurs in CPU mode…I try running it in K40，it works just fine…I choose the batch size =3 (reach the maximum memory)and change the classification type to 32 ,using the Camvid dataset,after 15000 iteration ,and it didn't converge...the result is not good-A-,i'm new in deep learning ,may be my parameter chioce is bad,(batch size)or the iteration is not enough...

**Comment by khufkens** [comment]
@SpecialM2580 How was this issue resolved I can't make heads nor tails from this message chain.

**Comment by codecolony** [comment]
@jorgyz Thanks, your solution seems to work!

**Comment by 5argon** [comment]
@jorgyz Thank you too, for those who came later and want to do that please note that there are 2 spots in the code that is throwing that fatal exception. You have to comment both out. And this seems to be a bug, since it is unlikely that just by switching from GPU to CPU will cause layer shapes to change...

**Comment by Jinming-Su** [comment]
@jorgyz Thank you too. The later person who would come across can comment out line 127 and line 128.
